server/src/main/java/io/crate/execution/engine/join/HashJoinBatchIterator.java

```diff
@@ -88,7 +100,8 @@ public class HashInnerJoinBatchIterator extends JoinBatchIterator<Row, Row, Row>
     private final ToIntFunction<Row> hashBuilderForLeft;
     private final ToIntFunction<Row> hashBuilderForRight;
     private final LongToIntFunction calculateBlockSize;
-    private final IntObjectHashMap<List<Object[]>> buffer;
+    private final IntObjectHashMap<Tuple<List<Object[]>, Boolean>> buffer;
```

Comment:
The marker creates a minimal overhead for the classical inner-hash join, because it is never used there. We could consider storing the marker in a seperate datastructure.

Reply:
We can use a record (or plain pojo class) with a boolean, which has a setter, so we also avoid creating new tuple object here: https://github.com/crate/crate/pull/16780/files#diff-22ffb32b68a7154a9e86e1d07e93a54bb2c9935b462ec8d3c073c0fd6903904aR284

Reply:
Additionally Let's have have a jmh benchmark run, for an inner join, to compare mem and CPU overhead of this, maybe it makes sense to have another class that maybe inherits from the Inner, to avoid adding the unnecessary overhead.